src/main/java/com/microsoft/sqlserver/jdbc/SecureStringUtil.java

```diff
@@ -88,24 +88,26 @@ private SecureStringUtil() throws SQLServerException {
     /**
      * Get encrypted value of given string
      * 
-     * @param str
+     * @param chars
      *        string to encrypt
      * 
      * @return encrypted string
      * 
      * @throws SQLServerException
      *         if error
      */
-    String getEncryptedString(String str) throws SQLServerException {
+    byte[] getEncryptedBytes(char[] chars) throws SQLServerException {
+        if (chars == null)
+            return null;
         SecureRandom random = new SecureRandom();
         random.nextBytes(iv);

```

Comment:
This is very fragile. Out of order encrypt-decrypt calls will not work because iv is stored in instance variable.